arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/section/call/CallSection.java

```diff
@@ -605,8 +605,14 @@ private void completeEoaSuccessWillRevert(Hub hub) {
   private void completeSmcFailureWillRevert(Hub hub) {
     scenarioFragment.setScenario(CALL_SMC_FAILURE_WILL_REVERT);
 
-    calleeThird = calleeSecondNew.deepCopy().setDeploymentNumber(hub);
-    calleeThirdNew = calleeFirst.deepCopy().setDeploymentNumber(hub);
+    if (isSelfCall()) {
+      calleeThird = callerSecondNew.deepCopy().setDeploymentNumber(hub);
+      calleeThirdNew = callerFirst.deepCopy().setDeploymentNumber(hub);
+    } else {
+      calleeThird = calleeSecondNew.deepCopy().setDeploymentNumber(hub);
+      calleeThirdNew = calleeFirst.deepCopy().setDeploymentNumber(hub);
+    }
+
```

Comment:
See https://github.com/Consensys/linea-specification/issues/83 for more explanations.